### PR TITLE
FIX NGSI patching macro subscription covering completely the string always cast to string for id/type

### PR DESCRIPTION
**Pull request by fgalan**
Issue #4462

### CODE REVIEW

File: doc/manuals/orion-api.md [hunk 1]
```diff
@@ -2253,8 +2253,10 @@ Some notes to take into account when using `ngsi`:
   * It cannot be used in the key part of JSON objects, i.e. `"${key}": 10` will not work
   * It cannot be used in the attribute `type`. Only in the `value` macro replacements can be done.
   * If the macro *covers completely the string where is used*, then the JSON nature of the attribute value
-    is taken into account. For instance, `"value": "${temperature}"` resolves to `"value": 10`
+    is taken into account. For instance, `"value": "${temperature}"` resolves to `"value": 10`.
```
Comment (fgalan) [inline doc/manuals/orion-api.md:2256]:
```suggestion
    is taken into account. For instance, `"value": "${temperature}"` resolves to `"value": 10`
```

**Review by danielvillalbamota — COMMENTED** [review]
LGTM

File: test/functionalTest/cases/4462_ngsi_patching_id_type_macro_always_stringfly/ngsi_patching_id_type_macro_always_stringfly.test [hunk 2]
```diff
@@ -0,0 +1,206 @@
...
+#
+# This file is part of Orion Context Broker.
+#
+# Orion Context Broker is free software: you can redistribute it and/or
+# modify it under the terms of the GNU Affero General Public License as
+# published by the Free Software Foundation, either version 3 of the
+# License, or (at your option) any later version.
+#
+# Orion Context Broker is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implie d warranty of
```
Comment (fgalan) [inline test/functionalTest/cases/4462_ngsi_patching_id_type_macro_always_stringfly/ngsi_patching_id_type_macro_always_stringfly.test:11]:
```suggestion
# but WITHOUT ANY WARRANTY; without even the implied warranty of
```

**Comment by mapedraza** [comment]
The following error appeared on CI:

```log
-----  Reset Metrics  -----
(2750_common_metrics/2750_reset_metrics.test) output not as expected
VALIDATION ERROR: input line:
                       "incomingTransactionResponseSize": 316,
does not match ref line:
                       "incomingTransactionResponseSize": 320,


================================================================================

----------- Failing tests ------------------
  o  2750_common_metrics/2750_reset_metrics.test
make: *** [makefile:91: functional] Error 1
Builder: revert makefile
Builder: functional test failed
Error: Process completed with exit code 1.
```

**Comment by fgalan** [comment]
@fisuda this PR does a small modification to doc/manuals/orion-api.md. It would be great if you could sync the Japanese translation, please. Thanks!

**Comment by fisuda** [comment]
I sent the PR #4472. Thanks.

**Pull request merged**